File: 백준/Bronze/14697. 방 배정하기/방 배정하기.py
# 방정원 A B C 1=< =< 50
#전체 학생수1=<  N=< 300
#빈침대 없이 출력시 1 불가능 0
# 서브태스크
# 완탐

A, B, C, N = map(int, input().split())
a = 0
b = 0
c = 0 # c 는 N-(a+b)
flag = False
for i in range(301):
    a = A * i
    for j in range(301):
        b = B * j
        for h in range(301):
            c = C * h
            if a + b + c == N:
                flag = True

if flag == True:
    print(1)
else:
    print(0)

File: 백준/Bronze/14697. 방 배정하기/방 배정하기.java
import java.util.*;
import java.io.*;

public class Main {
	static class Pair {
		int x;
		int y;

		public Pair(int x, int y) {
			this.x = x;
			this.y = y;
		}
	}
	static List<Pair> pairs;
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

		boolean flag = false;
		for (int i=0; i<=300; i++) {
			for (int j=0; j<=150; j++) {
				for (int h=0; h<=150; h++) {
					if (arr[0] * i + arr[1] * j + arr[2] * h == arr[3]) {
						flag = true;
						break;
					} else if (arr[0] * i + arr[1] * j + arr[2] * h > arr[3]) {
						break;
					}
				}
				if (flag) break;
			}
			if (flag) break;
		}

		System.out.print(flag ?  1 : 0);
	}
}
